contracts/vesting/VestingEscrowFactory.vy

```diff
@@ -24,80 +38,156 @@ interface VestingEscrowSimple:
+) -> uint256:
     """
-    @notice Deploy a new vesting contract
-    @dev Prior to deployment you must approve `amount` tokens
-    @param token ERC20 token being distributed
+    @notice Create a new YFI vest
+    @dev Prior to deployment you must approve `amount` YFI
     @param recipient Address to vest tokens for
     @param amount Amount of tokens being vested for `recipient`
     @param vesting_duration Time period (in seconds) over which tokens are released
     @param vesting_start Epoch time when tokens begin to vest
-    @param open_claim Switch if anyone can claim for `recipient`
-    @param owner Vesting contract owner
+    @param cliff_length Duration (in seconds) after which the first portion vests
+    @return Vest index
     """
     assert cliff_length <= vesting_duration  # dev: incorrect vesting cliff
     assert vesting_start + vesting_duration > block.timestamp  # dev: just use a transfer, dummy
     assert vesting_duration > 0  # dev: duration must be > 0
-    assert recipient not in [self, empty(address), token.address, owner] # dev: wrong recipient
+    assert recipient not in [self, empty(address), YFI.address, OWNER] # dev: wrong recipient
+    assert amount > 0
 
-    escrow: address = create_minimal_proxy_to(TARGET)
-    VestingEscrowSimple(escrow).initialize(
-        owner,
-        token,
+    idx: uint256 = self.num_vests
+    self.num_vests = idx + 1
+    self.pending_vests[idx] = Vest({
+        recipient: recipient,
+        amount: amount,
+        duration: vesting_duration,
+        start: vesting_start,
+        cliff: cliff_length
+    })
+    assert YFI.transferFrom(msg.sender, self, amount, default_return_value=True)
+
+    log VestingEscrowCreated(
+        msg.sender,
         recipient,
+        idx,
         amount,
         vesting_start,
-        vesting_start + vesting_duration,
-        cliff_length,
-        open_claim,
+        vesting_duration,
+        cliff_length
     )
 
-    # skip transferFrom and approve and send directly to escrow
-    assert token.transferFrom(msg.sender, escrow, amount, default_return_value=True)  # dev: funding failed
+    return idx
 
-    log VestingEscrowCreated(
+@external
+def deploy_vesting_contract(
+    idx: uint256,
+    token: address,
+    amount: uint256,
+    open_claim: bool = True,
+) -> (address, uint256):
+    """
+    @notice Deposit into a liquid locker and deploy a vesting contract
+    @dev Requires a vest. Requires the liquid locker being approved by the owner
+    @param idx Vest index
+    @param token Liquid locker token to deposit into
+    @param amount Amount of YFI to deposit
+    @param open_claim Whether anyone can claim 
+    @return Vest contract address, vested amount of liquid locker tokens
+    """
+    vest: Vest = self.pending_vests[idx]
+    assert msg.sender == vest.recipient
+
+    depositor: address = self.liquid_lockers[token]
+    assert depositor != empty(address)
+
+    self.pending_vests[idx].amount -= amount
```

Comment:
this is an important invariant to ensure a person cannot double spend its vesting and steal YFI vested from others allocation into their own vesting contract.

move this line before checking depositor since its a hard invariant that a user cannot double spend their vesting or steal amounts from another pending vest too, also small gas saving if it fails fast?

Reply:
I prefer to first do sanity checks and then modify state after. The gas savings shouldnt be relevant since under normal circumstances a user will never try to claim more than they have left